src/rust/core/server/src/threads/workers/multi.rs

```diff
@@ -118,6 +118,9 @@ where
             }
 
             WORKER_EVENT_TOTAL.add(events.iter().count() as _);
+            if events.iter().count() == self.nevent {
```

Comment:
instead of iterating twice here, let's have a temporary variable that we can use for both L120 and L121

Reply:
done